### PR TITLE
3771 review deps

### PR DESCRIPTION
**Pull request by meandavejustice**

### CODE REVIEW

File: addon/package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
   "author": "Mozilla (https://mozilla.org/)",
   "license": "MPL-2.0",
   "devDependencies": {
+    "babel-eslint": "^8.2.6",
```
Comment (fzzzy) [inline addon/package.json:16]: Remove the ^

File: frontend/src/app/components/RetireConfirmationDialog/index.js [hunk 2]
```diff
@@ -12,21 +12,17 @@ type RetireConfirmationDialogProps = {
...
-      this.modalContainer.focus();
-    }
+    if (this.modalContainer) this.modalContainer.focus();
   }
 
   render() {
     return (
       <div className="modal-container" tabIndex="0"
-        ref={modalContainer => { this.modalContainer = modalContainer; }}
+        ref={(modalContainer: any | null) => { this.modalContainer = modalContainer; }}
```
Comment (fzzzy) [inline frontend/src/app/components/RetireConfirmationDialog/index.js:25]: Doesn't "any" include "null"?

File: frontend/src/app/containers/ExperimentPage/index.js [hunk 3]
```diff
@@ -99,15 +83,15 @@ export class ExperimentDetail extends React.Component {
     }
   }
 
-  componentWillMount() {
+  UNSAFE_componentWillMount() {
```
Comment (fzzzy) [inline frontend/src/app/containers/ExperimentPage/index.js:86]: What's up with UNSAFE?

File: frontend/src/app/containers/ExperimentPage/index.js [hunk 4]
```diff
@@ -99,15 +83,15 @@ export class ExperimentDetail extends React.Component {
...
     this.checkCookies();
   }
 
   componentDidUpdate() {
     this.checkCookies();
   }
 
-  componentWillReceiveProps(nextProps: ExperimentDetailProps) {
+  UNSAFE_componentWillReceiveProps(nextProps: ExperimentDetailProps) {
```
Comment (fzzzy) [inline frontend/src/app/containers/ExperimentPage/index.js:94]: Ditto

File: frontend/src/app/containers/ExperimentPage/index.js [hunk 5]
```diff
@@ -383,15 +368,9 @@ export class ExperimentDetail extends React.Component {
       return;
     }
 
-    let progressButtonWidth;
```
Comment (fzzzy) [inline frontend/src/app/containers/ExperimentPage/index.js:386]: Nice to get rid of this ugly hack

File: frontend/src/app/containers/UpgradeWarningPage/index.js [hunk 6]
```diff
@@ -16,10 +16,9 @@ type UpgradeWarningProps = {
   isProdHost: Boolean
 }
 
-export default class UpgradeWarning extends React.Component {
-  props: UpgradeWarningProps
+export default class UpgradeWarning extends Component<UpgradeWarningProps> {
 
-  componentWillMount() {
+  UNSAFE_componentWillMount() {
```
Comment (fzzzy) [inline frontend/src/app/containers/UpgradeWarningPage/index.js:21]: Ditto

**Review by fzzzy — CHANGES_REQUESTED** [review]
I'm super excited about this. Great job! I know it has been hard, but I think it will be worth it.

File: frontend/src/app/containers/ExperimentPage/index.js (hunk 4 above)
Comment (meandavejustice) [reply frontend/src/app/containers/ExperimentPage/index.js:94]: React is deprecating these calls, it was showing up in the linter

**Pull request closed without merging**